### DISCUSSION TITLE
Running code in a service after a response has been sent.

### DISCUSSION THREAD

**Discussion by Aurora2500** [actix/actix-web#2839] [Q&A]
I'm making a webserver that deals with websockets. It will need to run some code after it knows that a websocket connection has been made, so it needs to wait for the service to send a response to complete the protocol change handshake. But all the API I see for making services relies on returning a type for the HTTP response, making it impossible to run code after. Is there any way I can reply with a Response to a request and then run some more code after?

**Comment by robjtede** [comment]
You could use `tokio::spawn` to schedule more work to be done in the near future. Though, in general, "after it knows that a websocket connection has been made" cannot be assumed until the client has sent a WebSocket message back. I'd recommend leaning more on WebSocket message passing and doing some sort of initialization handshake within the WS, running your custom code during that.
